analyzers/tests/SonarAnalyzer.UnitTest/Extensions/SyntaxNodeExtensionsTest.cs

```diff
@@ -410,6 +412,82 @@ private void Method()
             ReferenceEquals(cfg1, cfg2).Should().BeFalse("Different compilations should not reuse cache. They do not share semantic model and symbols.");
         }
 
+        [DataTestMethod]
+        // Tuple. From right to left.
+        [DataRow("(var a, (x, var b)) = (0, ($$x++, 1));", "x")]
+        [DataRow("(var a, (x, var b)) = (0, $$(1, 2));", "(x, var b)")]
+        [DataRow("(var a, (var b, var c, var d), var e) = (0, (1, 2, $$3), 4);", "var d")]
+        // Tuple. From left to right
+        [DataRow("(var a, (x, $$var b)) = (0, (x++, 1));", "1")]
+        [DataRow("(var a, (var b, var c, $$var d), var e) = (0, (1, 2, 3), 4);", "3")]
+        [DataRow("(var a, $$(var b, var c)) = (0, (1, 2));", "(1, 2)")]
+        // Designation. From right to left.
+        [DataRow("var (a, (b, c)) = (0, (1, $$2));", "c")]
+        [DataRow("var (a, (b, c)) = (0, $$(1, 2));", "(b, c)")]
+        [DataRow("var (a, (b, _)) = (0, (1, $$2));", "_")]
+        [DataRow("var (a, _) = (0, ($$1, 2));", null)]
+        [DataRow("var (a, _) = (0, $$(1, 2));", "_")]
+        [DataRow("var _ = ($$0, (1, 2));", null)]
+        [DataRow("_ = ($$0, (1, 2));", null)]
+        // Designation. From left to right
+        [DataRow("var (a, (b, $$c)) = (0, (1, 2));", "2")]
+        [DataRow("var (a, $$(b, c)) = (0, (1, 2));", "(1, 2)")]
+        [DataRow("var (a, (b, $$_)) = (0, (1, 2));", "2")]
+        [DataRow("var (a, $$_) = (0, (1, 2));", "(1, 2)")]
+        // Unaligned tuples. From left to right.
+        [DataRow("(var a, var b) = ($$0, 1, 2);", null)]
+        [DataRow("(var a, var b) = (0, 1, $$2);", null)]
+        [DataRow("(var a, var b) = (0, (1, $$2));", null)]
+        [DataRow("(var a, (var b, var c)) = (0, $$1);", "(var b, var c)")] // Syntacticly correct
+        // Unaligned tuples. From right to left.
+        [DataRow("(var a, var b, $$var c) = (0, (1, 2));", null)]
+        [DataRow("(var a, (var b, $$var c)) = (0, 1, 2);", null)]
+        // Unaligned designation. From right to left.
+        [DataRow("var (a, (b, c)) = (0, (1, $$2, 3));", null)]
+        [DataRow("var (a, (b, c)) = (0, (1, ($$2, 3)));", null)]
+        [DataRow("var (a, (b, c, d)) = (0, (1, $$2));", null)]
+        // Unaligned designation. From left to right .
+        [DataRow("var (a, (b, $$c)) = (0, (1, 2, 3));", null)]
+        [DataRow("var (a, (b, ($$c, d))) = (0, (1, 2));", null)]
+        [DataRow("var (a, (b, $$c, d)) = (0, (1, 2));", null)]
+        // Mixed. From right to left.
+        [DataRow("(var a, var (b, c)) = (1, ($$2, 3));", "b")]
+        [DataRow("(var a, var (b, (c, (d, e)))) = (1, (2, (3, (4, $$5))));", "e")]
+        // Mixed. From left to right.
+        [DataRow("(var a, var ($$b, c) )= (1, (2, 3));", "2")]
+        [DataRow("(var a, var (b, (c, (d, $$e)))) = (1, (2, (3, (4, 5))));", "5")]
```

Comment:
can you add a test with tuple types in ParenthesizedVariableDesignation? or is it not allowed?